### DISCUSSION TITLE
Photos from shared albums not downloaded to mac

### DISCUSSION THREAD

**Discussion by Se7enair** [RhetTbull/osxphotos#720] [Q&A]
I have a few shared albums in the icloud. But it seems that the photos.app on mac will not download it locally.
Everything else seems to be downloaded, except shared albums.

I use Monterey 12.4. I know this is not related to osxphotos, but maybe somebody can help.

**Comment by RhetTbull** [comment]
Do the albums show up in Photos at all? (That is, you see them in the shared album list but photos are not downloaded?) Or are they completely absent? 

One option (though I wouldn't do this just yet without knowing more about your situation) is to tell photos to repair the library. Launch Photos while holding down Command + Option. This forces Photos to rebuild the database used to track all assets in the library.

**Reply by Se7enair** [reply]
The albums show up in Photos.
I tried to deactive shared albums and reactivated it. No improvements.

For now, I use --download-missing. Earlier I mistyped a underline and didnt get this option to work.

**Reply by Se7enair** [reply]
For example, if there is no photo locally saved, osxphotos will download every single photo. Good to save local space. 
But how about a photo which is already downloaded? It wont be downloaded again right? So a second export, with same options and no change in the photos.app, shouldnt download any photo?

**Reply by RhetTbull** [reply]
> But how about a photo which is already downloaded? It wont be downloaded again right? 

In most cases, this is true--the photo won't be downloaded again if you have Photos set to "Download originals to this Mac" in the Preferences.  But there are exceptions that osxphotos cannot control.  Shared photos are not always kept by Photos, even when "Download originals to this Mac" is checked.  I've seen other cases where Photos does not retain the downloaded image in the library even though "Download originals" is checked.

**Comment by Se7enair** [comment]
Perfect. I emptied the whole local library and it downloaded every picture I wanted.
